arroyo/processing/processor.py

```diff
@@ -17,13 +17,48 @@
 
 logger = logging.getLogger(__name__)
 
-LOG_THRESHOLD_TIME = 10  # In seconds
+LOG_THRESHOLD_TIME = 5  # In seconds
 
 
 class InvalidStateError(RuntimeError):
     pass
 
 
+class MetricsBuffer:
```

Comment:
I think this can be useful in many other contexts. It should be the default way to record metrics in a variable throughput environment. You could make it generic and reusable. Not a blocker.

Reply:
Yes. But I don't want to make it generic until we actually need it to be. It's not part of the public API so will not be a big deal to update later if we decide to reuse.